File: AgentApp/Backend/src/main/java/jvn/RentACar/model/Comment.java
package jvn.RentACar.model;

import jvn.RentACar.enumeration.CommentStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String text;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private User sender;

    @Column(nullable = false)
    private CommentStatus status;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private RentInfo rentInfo;
}

File: MainApp/ZuulAndFrontend/Frontend/src/app/model/comment.ts
import { UserInfo } from "./userInfo";
import { RentInfo } from "./rentInfo";

export class Comment {
  id: number;
  text: string;
  sender: UserInfo;
  status: string;
  constructor(text: string, sender: UserInfo, status?: string, id?: number) {
    this.text = text;
    this.status = status;
    this.id = id;
    this.sender = sender;
  }
}
